### PR TITLE
Polkadot v0.9.18

### PR DESCRIPTION
**Pull request by jasl**
https://github.com/paritytech/substrate/tree/polkadot-v0.9.18 is out, I guess we should start to do the update, and `ink 3.0-rc9` too

- [x] Update dependencies to `polkadot-v0.9.18` branch
- [x] Update `ink` to `3.0-rc9`
- [x] Update `scale-codec` to `3.0`, `scale-info` to 2.0
- [x] Rebase `subxt`
- [x] Fix compiling errors

Because of Polkadot 0.9.18 changes much, I'm concerning current mining suites won't work, but we have enough time to test.

### CODE REVIEW

**Comment by jasl** [comment]
```
error[E0425]: cannot find function `memory_teardown` in module `sandbox`
  --> substrate/primitives/sandbox/src/host_executor.rs:57:12
   |
57 |         sandbox::memory_teardown(self.memory_idx);
   |                  ^^^^^^^^^^^^^^^ not found in `sandbox`

error[E0425]: cannot find function `memory_new` in module `sandbox`
  --> substrate/primitives/sandbox/src/host_executor.rs:74:18
   |
74 |         match sandbox::memory_new(initial, maximum) {
   |                        ^^^^^^^^^^ not found in `sandbox`

error[E0425]: cannot find function `memory_get` in module `sandbox`
  --> substrate/primitives/sandbox/src/host_executor.rs:82:13
   |
82 |             sandbox::memory_get(self.handle.memory_idx, offset, buf.as_mut_ptr(), buf.len() as u32);
   |                      ^^^^^^^^^^ not found in `sandbox`

error[E0425]: cannot find function `memory_set` in module `sandbox`
  --> substrate/primitives/sandbox/src/host_executor.rs:91:25
   |
91 |         let result = sandbox::memory_set(
   |                               ^^^^^^^^^^ not found in `sandbox`

error[E0425]: cannot find function `instantiate` in module `sandbox`
   --> substrate/primitives/sandbox/src/host_executor.rs:223:25
    |
223 |         let result = sandbox::instantiate(
    |                               ^^^^^^^^^^^ not found in `sandbox`

error[E0425]: cannot find function `invoke` in module `sandbox`
   --> substrate/primitives/sandbox/src/host_executor.rs:249:25
    |
249 |         let result = sandbox::invoke(
    |                               ^^^^^^ not found in `sandbox`

error[E0425]: cannot find function `get_global_val` in module `sandbox`
   --> substrate/primitives/sandbox/src/host_executor.rs:270:12
    |
270 |         sandbox::get_global_val(self.instance_idx, name)
    |                  ^^^^^^^^^^^^^^ not found in `sandbox`

error[E0425]: cannot find function `instance_teardown` in module `sandbox`
   --> substrate/primitives/sandbox/src/host_executor.rs:276:12
    |
276 |         sandbox::instance_teardown(self.instance_idx);
    |                  ^^^^^^^^^^^^^^^^^ not found in `sandbox`

For more information about this error, try `rustc --explain E0425`.
```

**Comment by jasl** [comment]
```
error[E0432]: unresolved import `core::prelude::v1::llvm_asm`
  --> /home/jasl/Workspaces/Remote/phala-blockchain/teaclave-sgx-sdk/sgx_tstd/src/prelude/v1.rs:49:64
   |
49 |     format_args_nl, include, include_bytes, include_str, line, llvm_asm, log_syntax, module_path,
   |                                                                ^^^^^^^^ no `llvm_asm` in `prelude::v1`

error[E0432]: unresolved import `core::stream`
   --> /home/jasl/Workspaces/Remote/phala-blockchain/teaclave-sgx-sdk/sgx_tstd/src/lib.rs:216:9
    |
216 | pub use core::stream;
    |         ^^^^^^^^^^^^ no `stream` in the root

error[E0432]: unresolved import `core::time::FromSecsError`
  --> /home/jasl/Workspaces/Remote/phala-blockchain/teaclave-sgx-sdk/sgx_tstd/src/time.rs:61:9
   |
61 | pub use core::time::FromSecsError;
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^ no `FromSecsError` in `time`

error[E0432]: unresolved import `core::llvm_asm`
   --> /home/jasl/Workspaces/Remote/phala-blockchain/teaclave-sgx-sdk/sgx_tstd/src/lib.rs:298:88
    |
298 |     env, file, format_args, format_args_nl, include, include_bytes, include_str, line, llvm_asm,
    |                                                                                        ^^^^^^^^ no `llvm_asm` in the root

error[E0412]: cannot find type `FromSecsError` in module `core::time`
   --> /home/jasl/Workspaces/Remote/phala-blockchain/teaclave-sgx-sdk/sgx_tstd/src/error.rs:573:28
    |
573 | impl Error for core::time::FromSecsError {}
    |                            ^^^^^^^^^^^^^ not found in `core::time`
```

**Comment by jasl** [comment]
```
cp ./target/release/libenclaveapp.a ../lib/libenclave.a
make[1]: Leaving directory '/home/jasl/Workspaces/Remote/phala-blockchain/standalone/pruntime/enclave'
CC   <=  enclave/Enclave_t.c
/usr/bin/ld: ./lib/libenclave.a(std-1b64d5fe7a3c3d7f.std.b9945fc4-cgu.0.rcgu.o): in function `std::sys::unix::os::glibc_version':
/rustc/68369a041cea809a87e5bd80701da90e0e0a4799//library/std/src/sys/unix/os.rs:639: undefined reference to `gnu_get_libc_version'
collect2: error: ld returned 1 exit status
make: *** [Makefile:184: enclave/enclave.so] Error 1
```

File: standalone/node/src/cli.rs [hunk 1]
```diff
@@ -16,38 +16,34 @@
...
+
 	#[allow(missing_docs)]
-	#[structopt(flatten)]
-	pub run: RunCmd,
-	#[allow(missing_docs)]
-	#[structopt(long, help = "Custom block duration in milliseconds (only useful with --dev)")]
+	#[clap(flatten)]
+	pub run: sc_cli::RunCmd,
+
+	/// Custom block duration in milliseconds (only useful with --dev)
```
Comment (h4x3rotab) [inline standalone/node/src/cli.rs:30]: Will it show the help message in the cli or just docstring?

**Review by h4x3rotab — APPROVED** [review]

File: standalone/node/src/cli.rs (hunk 1 above)
Comment (jasl) [reply standalone/node/src/cli.rs:30]:
```
--block-millisecs <BLOCK_MILLISECS>
            Custom block duration in milliseconds (only useful with --dev)
```

yes!

**Pull request merged**